SystemInfo/SystemInfoCollectorRegistry.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ * File containing the SystemInfoCollectorRegistry interface.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzSupportToolsBundle\SystemInfo;
+
+/**
+ * A registry of SystemInfoCollectors.
+ */
+interface SystemInfoCollectorRegistry
+{
+    /**
+     * Adds SystemInfoCollectors, each with an identifier string, to the registry.
+     *
+     * @param \EzSystems\EzSupportToolsBundle\SystemInfo\Collector\SystemInfoCollector[] $items A hash of SystemInfoCollectors, with identifier string as key.
+     */
+    public function addItems(array $items);
```

Comment:
Not really following the discussion here.

But another point; what about renaming it `setItems`, in real life you only need to set it once right? And if someone, even tests needs to append, then they can use `getItems` to append and then set. This will avoid the need to do array merge also.